### PR TITLE
Detect if VPN is connected and Normandy Admin is available

### LINKED ISSUE #297: Add VPN availability to state
If a user doesn't have VPN access to Normandy Admin, Delivery Console should fall back to using the read only public Normandy API. In order to do this, we need to detect if the VPN is available. This issue covers detecting this, and putting it in the state. It doesn't cover automatically routing requests to different backends.

### PR DESCRIPTION
**Pull request by mythmon**
* Added a new state subtree, `network`.
* Move the `requests` state subtree to `network.requests`
* Added `network.availability` state, with a subkey `normandyAdmin`
* Make a canary request to Normandy Admin at startup, and mark it as unavailable if that fails
* Show this data in the header as a warning icon with a tooltip

Currently, the admin's availability is only checked at startup, and requires a refresh to update. In the future we could be smarter and do ongoing checks, including responding to failed requests to the admin. For now I think checking at startup is good enough. I looked into automatically detecting network changes, but [that API isn't available in Firefox](https://developer.mozilla.org/en-US/docs/Web/API/NetworkInformation#Browser_compatibility).

There might be some things that could be done better with the state handling, specifically around the handling of the pending state. It works ok now, but if we scale to other backend servers that may or may not be available, we might want a richer state than just a boolean.

I expect the UI here to be temporary, so I didn't put a lot of work into it. It is visible enough if you look for it, and I hope that soon we'll have a automatic fallback, so a big scary warning won't be needed.

Fixes #297.

### CODE REVIEW

**Comment by peterbe** [comment]
I didn't review the code but skimmed through it and it looks really good. I like the detail of calling it `network` by the way. Great work @mythmon !

**Comment by mythmon** [comment]
> ```REACT_APP_NORMANDY_ADMIN_API_ROOT_URL=https://normandy-admin.prod.mozaws.net/api/```

You'll probably want to use https://normandy-admin-bearer.stage.mozaws.net. Since `normandy-admin` will redirect you over to auth0, it isn't ever going to work, so VPN will never be detected. I tested this by turning my local server on and off.

> But the loading spinner in the tab never stops

I expect that will be true no matter what, since I didn't change any of the data fetching behavior.

File: src/components/common/ShieldIdenticon.js [hunk 1]
```diff
@@ -4,7 +4,7 @@ import cx from 'classnames';
 import React from 'react';
 import PropTypes from 'prop-types';
 
-import { NORMANDY_API_ROOT_URL } from 'console/settings';
+import { NORMANDY_ADMIN_API_ROOT_URL } from 'console/settings';
```
[the remark this answers is not in the document]
Comment (mythmon) [reply src/components/common/ShieldIdenticon.js]: For now I did a simple replacement to convert `NORMANDY_API_ROOT_URL` to `NORMANDY_ADMIN_API_ROOT_URL` everywhere, saving any use of the read-only server to #298.

**Comment by rehandalal** [comment]
I was able to figure out the deluge of requests problem. It was due to a misconfiguration on my part. Still don't see a status message though.

File: src/components/data/QueryNormandyAdmin.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  {
+    detectNormandyAdmin,
+  },
+)
+export default class QueryNormandyAdmin extends React.PureComponent {
+  static propTypes = {
+    detectNormandyAdmin: PropTypes.func.isRequired,
+  };
+
+  async componentWillMount() {
```
Comment (rehandalal) [inline src/components/data/QueryNormandyAdmin.js]: this needs to change to `componentDidMount` which is the failure when merging.

**Comment by mythmon** [comment]
Fixed the lint errors, and found a proptype error too, that I also fixed.

bors r= @rehandalal

**Pull request merged**